projects/js-packages/publicize-components/src/social-store/actions/share-status.ts

```diff
@@ -70,3 +70,78 @@ export function openShareStatusModal() {
 export function closeShareStatusModal() {
 	return toggleShareStatusModal( false );
 }
+
+type IsRequestComplete = ( options: {
+	lastTimestamp: number;
+	postShareStatus: PostShareStatus;
+} ) => boolean;
+
+/**
+ * Default implementation to check if the request is complete.
+ *
+ * @param {IsRequestComplete} options - Options.
+ *
+ * @return {boolean} - Whether the request is complete.
+ */
+export const defaultIsRequestComplete: IsRequestComplete = ( {
+	lastTimestamp,
+	postShareStatus,
+} ) => {
+	// If the last timestamp is present, check if there are newer timestamps.
+	// otherwise check if we have any shares.
+	return lastTimestamp
+		? postShareStatus.shares.some( share => share.timestamp > lastTimestamp )
+		: postShareStatus.shares.length > 0;
+};
+
+type PollForPostShareStatusOptions = {
+	postId?: number;
+	timeout?: number;
+	isRequestComplete?: IsRequestComplete;
+	pollingInterval?: number;
+};
+
+const ONE_MINUTE_IN_MS = 60 * 1000;
+
+const POLLING_INTERVAL = 3 * 1000; // milliseconds
+
+/**
+ * Poll for share status.
+ *
+ * @param {PollForPostShareStatusOptions} options - Options.
+ *
+ * @return {Promise<void>} - Function to start polling.
+ */
+export function pollForPostShareStatus( {
+	pollingInterval = POLLING_INTERVAL,
+	postId,
+	isRequestComplete = defaultIsRequestComplete,
+	timeout = ONE_MINUTE_IN_MS,
+}: PollForPostShareStatusOptions = {} ) {
+	return async function ( { dispatch, select } ) {
+		const startedAt = Date.now();
```

Comment:
Worth adding a try-catch block here to avoid an editor crash. We are doing a few things within this function, so could be could to protect against a crash.

Reply:
There is nothing in the function that throws. So, it will be unnecessary.

Also, this action is dispatched only behind the feature flag. So, it should be fine.